dolphinscheduler-task-plugin/dolphinscheduler-task-sql/src/main/java/org/apache/dolphinscheduler/plugin/task/sql/SqlTask.java

```diff
@@ -306,13 +319,31 @@ private String executeUpdate(Connection connection, List<SqlBinds> statementsBin
         int result = 0;
         for (SqlBinds sqlBind : statementsBinds) {
             try (PreparedStatement statement = prepareStatementAndBind(connection, sqlBind)) {
+
+                sqlLogListener(statement);
+
                 result = statement.executeUpdate();
                 logger.info("{} statement execute update result: {}, for sql: {}", handlerType, result, sqlBind.getSql());
             }
         }
         return String.valueOf(result);
     }
 
+    /**
+     * sql log listener
+     *
+     * @param statement statement
+     * @throws SQLException sqlException
+     */
+    private void sqlLogListener(PreparedStatement statement) throws SQLException {
+        if (DbType.HIVE == DbType.valueOf(sqlParameters.getType())) {
+            logger.info("The current sql task type is [{}], will start a demon thread to resolve the appIds", DbType.HIVE);
+
+            HiveSqlLogThread queryThread = new HiveSqlLogThread(statement, logger,taskExecutionContext);
```

Comment:
The purpose here is to start the thread to print the progress log of MR, so that there is a progress log of HIVE SQL execution in the worker log file, and then we can obtain appids using the LogUtils.getAppids() method.

Reply:
Is it possible to parse all the logs at once after execution?

Reply:
The screenshot of the print log is as follows:
<img width="821" alt="image" src="https://user-images.githubusercontent.com/35831367/189870359-16ba2b67-ece2-405a-97a7-4ffd7c8d37da.png">

In SqlTask.class ,Parse all logs and get appid as follows:
<img width="869" alt="image" src="https://user-images.githubusercontent.com/35831367/189877917-fe6f4e98-2bc3-406e-8e2b-009b5e2df76d.png">

Reply:
@caishunfeng You can review my latest code